compiler/ast/parser.nim

```diff
@@ -1309,32 +1298,32 @@ proc primary(p: var Parser, mode: PrimaryMode): PNode =
     if mode != pmSkipSuffix:
       result = p.primarySuffix(result, baseInd, mode)
 
-proc binaryNot(p: var Parser; a: PNode): PNode =
+proc binaryNot(p: var Parser; a: ParsedNode): ParsedNode =
   if p.tok.tokType == tkNot:
-    let notOpr = newIdentNodeP(p.tok.ident, p)
+    let notOpr = p.newNode(nkIdent, p.tok)
     p.getTok
     p.optInd(notOpr)
-    newTreeI(nkInfix, p.lineInfo, [notOpr, a, parseExpr(p)])
+    return p.newNode(nkInfix, @[notOpr, a, parseExpr(p)])
   else:
-    a
+    return a
```

Comment:
ok, I will change it to the `result=`, I don't have any specific preference for the `return` over `result=`

Reply:
I like to assign to result once, or at least as few times as possible, and try to avoid return.

Effectively the code should be really linear and a simple fall through. Then the reader can jump to the end and that's also the logical end. With mid-proc returns the logical end and written end don't match. All in an effort to let people (including future me) quickly skim code, get the gist of things, and keep flow.